src/meta/src/rpc/ddl_controller.rs

```diff
@@ -614,6 +620,45 @@ where
             self.source_manager
                 .unregister_sources(vec![source_id])
                 .await;
+            // drop replication slot for pg sources
+            if source.get_properties().get("connector") == Some(&POSTGRES_CDC_CONNECTOR.to_string())
```

Comment:
Just like what I commented in https://github.com/risingwavelabs/risingwave/pull/10455#discussion_r1236458142, do we have any other mechanisms to make sure that the slot will be cleaned if any failover occurs? Cc. @StrikeW

Reply:
> do we have any other mechanisms to make sure that the slot will be cleaned if any failover occurs?

I think not right now. Meta completely relies on the etcd transaction to provide atomicity, I am considering whether we can also leverage etcd to provide atomicity for external resources cleanup.